Client/Frontend/Browser/Search/SearchViewController.swift

```diff
@@ -419,52 +416,58 @@ class SearchViewController: SiteTableViewController, LoaderListener {
+          RecentSearch.addItem(type: .text, text: searchQuery, websiteUrl: nil)
         }
-        searchDelegate?.searchViewController(self, didSelectURL: url)
-      }
-    case .bookmarksAndHistory:
-      let site = data[indexPath.row]
-      if let url = URL(string: site.url) {
-        searchDelegate?.searchViewController(self, didSelectURL: url)
-      }
-    case .findInPage:
-      let localSearchQuery = searchQuery.lowercased()
-      searchDelegate?.searchViewController(self, shouldFindInPage: localSearchQuery)
+        searchDelegate?.searchViewController(self, didSubmit: searchQuery)
+      case .searchSuggestionsOptIn: return
+      case .searchSuggestions:
+        // Assume that only the default search engine can provide search suggestions.
+        let engine = searchEngines?.defaultEngine()
+        let suggestion = suggestions[indexPath.row]
+        
+        var url = URIFixup.getURL(suggestion)
+        if url == nil {
+          url = engine?.searchURLForQuery(suggestion)
+        }
+        
+        if let url = url {
+          if !PrivateBrowsingManager.shared.isPrivateBrowsing {
+            RecentSearch.addItem(type: .website, text: suggestion, websiteUrl: url.absoluteString)
+          }
+          searchDelegate?.searchViewController(self, didSelectURL: url)
+        }
+      case .openTabsAndHistoryAndBookmarks:
+        let site = data[indexPath.row]
+        
+
+        if let url = URL(string: site.url) {
+          if site.siteType == .tab {
+            searchDelegate?.searchViewController(self, didSelectOpenTabURL: url)
+          } else {
+            searchDelegate?.searchViewController(self, didSelectURL: url)
+          }
+        }
+      case .findInPage:
+        let localSearchQuery = searchQuery.lowercased()
+        searchDelegate?.searchViewController(self, shouldFindInPage: localSearchQuery)
     }
   }
 
   override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
     if let currentSection = availableSections[safe: indexPath.section] {
       switch currentSection {
-      case .quickBar:
-        return super.tableView(tableView, heightForRowAt: indexPath)
-      case .searchSuggestionsOptIn:
-        return 100.0
-      case .searchSuggestions:
-        return 44.0
-      case .bookmarksAndHistory:
-        return super.tableView(tableView, heightForRowAt: indexPath)
-      case .findInPage:
-        return super.tableView(tableView, heightForRowAt: indexPath)
+        case .quickBar:
+          return super.tableView(tableView, heightForRowAt: indexPath)
+        case .searchSuggestionsOptIn:
+          return 100.0
+        case .searchSuggestions:
+          return 44.0
+        case .openTabsAndHistoryAndBookmarks:
+          return super.tableView(tableView, heightForRowAt: indexPath)
+        case .findInPage:
+          return super.tableView(tableView, heightForRowAt: indexPath)
```

Comment:
seems like all the switch indentations are bit off :(

Reply:
Fix at https://github.com/brave/brave-ios/pull/5157/commits/f1acef3104359c6e8db06fa7cddd1ab1a1deebdc